samples/js/flickr/slurp/src/flickr.js

```diff
@@ -0,0 +1,131 @@
+// @flow
+
+// Copyright 2016 Attic Labs, Inc. All rights reserved.
+// Licensed under the Apache License, version 2.0:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+import flickrAPI from 'flickr-oauth-and-upload';
+import readline from 'readline';
+import {
+  invariant,
+  jsonToNoms,
+  Struct,
+} from '@attic/noms';
+
+export default class Flickr {
+  apiKey: string;
+  apiSecret: string;
+  accessToken: string;
+  accessTokenSecret: string;
+
+  constructor(apiKey: string, apiSecret: string,
+              accessToken: string = '', accessTokenSecret: string = '') {
+    this.apiKey = apiKey;
+    this.apiSecret = apiSecret;
+    this.accessToken = accessToken;
+    this.accessTokenSecret = accessTokenSecret;
+  }
+
+  async authenticate(): Promise<void> {
+    const [token, secret, url] = await getAuthToken(this.apiKey, this.apiSecret);
+    const verificationCode = await promptForVerificationCode(url);
+    // $FlowIssue: Flow does not understand destructuring assignment.
+    [this.accessToken, this.accessTokenSecret] =
+        await this.getAccessToken(token, secret, verificationCode);
+  }
+
+  getAccessToken(oauthToken: string, oauthTokenSecret: string, oauthVerifier: string):
```

Comment:
Could you:
 * make this private
 * have it set `this.accessToken` and `this.accessTokenSecret` itself, rather than assigning from `authenticate`.
 * not return anything (well, Promise<void> I suppose)
 * rename to `initAccessToken` or something